alf/algorithms/agent.py

```diff
@@ -329,47 +342,24 @@ def train_step(self, exp: Experience, state):
                 "`action_distribution`, which is required by "
                 "`enforce_entropy_target`")
             et_step = self._entropy_target_algorithm.train_step(
-                rl_step.info.action_distribution, step_type=exp.step_type)
+                (rl_step.info.action_distribution, time_step.step_type))
             info = info._replace(entropy_target=et_step.info)
 
         return AlgStep(output=rl_step.output, state=new_state, info=info)
 
-    def calc_training_reward(self, external_reward, info: AgentInfo):
```

Comment:
We should be very careful with this calc_training_reward function, especially in the off-policy training case.